nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBRepository.java

```diff
@@ -294,9 +318,12 @@ public String generateNewId() {
     public void createState(State state) {
         Document doc = converter.stateToBson(state);
         log.trace("MongoDB: CREATE {}: {}", doc.get(idKey), doc);
-        coll.insertOne(doc);
-        // TODO dupe exception
-        // throw new DocumentException("Already exists: " + id);
+        try {
+            coll.insertOne(doc);
+        } catch (DuplicateKeyException dke) {
+            log.trace("MongoDB:    -> DUPLICATE KEY: {}", doc.get(idKey));
+            throw new ConcurrentUpdateException(dke);
```

Comment:
Why not using `DocumentExistsException`? As it, we will be consistent with the `CoreSession#move` API.

Reply:
The idea behind this test is to ensure that the MongoDB will throw a concurrentUpdateException when a unique index is defined (here on parent and his child name). When we insert the call directly the backend (mongo) there is no check in Nuxeo code before insert. On other hand when we move there is a check on the name it's why on move i just kept the test and i check on `DocumentExistsException`